lib/internal/util.js

```diff
@@ -356,16 +356,17 @@ function isInsideNodeModules() {
 
   // Iterate over all stack frames and look for the first one not coming
   // from inside Node.js itself:
-  for (const frame of stack) {
-    const filename = frame.getFileName();
-    // If a filename does not start with / or contain \,
-    // it's likely from Node.js core.
-    if (!/^\/|\\/.test(filename))
-      continue;
-    return kNodeModulesRE.test(filename);
+  if (Array.isArray(stack)) {
```

Comment:
Is this sufficient to fix the original bug with ts-node? That was triggering at original line 360 with frame existing but getFileName being undefined, which is inside the for loop.  That seems to imply to me that it would not be fixed by an Array.isArray check. I'm not sure how to grab this to confirm that though.

Reply:
Yeah, it's because if stack is not an array then it's a string so it iterates on individual characters which obviously don't have `getFileName`.

Reply:
It *might* be good to throw in an extra check for the existence of `getFileName` as a `function`, just to be extra safe.

Reply:
Hmmm... but we do want to minimize how much work we do here. There are two possibilities here:

1. We call our own `prepareStackTrace` which returns an Array with the correct data.

2. We are already in an error handling mode and just get the default stack trace which is a string, not an array.

There's no real alternative here. We can't end up calling a user-provided `prepareStackTrace` or something similar. I did think about it before making this check.